### PR TITLE
CRDs for AllocationOverflow

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature

> /kind hotfix

**What this PR does / Why we need it**:

Implementation of yaml and backing Go code for the CRDs for supporting Fleet Allocation Overflow tracking.

**Which issue(s) this PR fixes**:

Work on #2682

**Special notes for your reviewer**:

Next will be controller code and e2e tests, docs are also incoming.

### CODE REVIEW

**Review by gongmax — APPROVED** [review]
LGTM

File: pkg/apis/agones/v1/common_test.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
+// Copyright 2022 Google LLC All Rights Reserved.
```
Comment (aimuz) [inline pkg/apis/agones/v1/common_test.go:1]: 2023?

Comment (markmandel) [reply pkg/apis/agones/v1/common_test.go:1]: Good catch!

Comment (markmandel) [reply pkg/apis/agones/v1/common_test.go:1]: Fixed.

**Comment by markmandel** [comment]
@chiayi this mean anything to you? Saw this flake in my e2e tests.

```
neric-1.24: Step #1 - "e2e-feature-gates": --- FAIL: TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown (0.15s)
generic-1.24: Step #1 - "e2e-feature-gates": panic: runtime error: invalid memory address or nil pointer dereference [recovered]
generic-1.24: Step #1 - "e2e-feature-gates": 	panic: runtime error: invalid memory address or nil pointer dereference
generic-1.24: Step #1 - "e2e-feature-gates": [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x20f9f70]
generic-1.24: Step #1 - "e2e-feature-gates": 
generic-1.24: Step #1 - "e2e-feature-gates": goroutine 9 [running]:
generic-1.24: Step #1 - "e2e-feature-gates": testing.tRunner.func1.2({0x22a4200, 0x340b020})
generic-1.24: Step #1 - "e2e-feature-gates": 	/usr/local/go/src/testing/testing.go:1396 +0x372
generic-1.24: Step #1 - "e2e-feature-gates": testing.tRunner.func1()
generic-1.24: Step #1 - "e2e-feature-gates": 	/usr/local/go/src/testing/testing.go:1399 +0x5f0
generic-1.24: Step #1 - "e2e-feature-gates": panic({0x22a4200, 0x340b020})
generic-1.24: Step #1 - "e2e-feature-gates": 	/usr/local/go/src/runtime/panic.go:890 +0x262
generic-1.24: Step #1 - "e2e-feature-gates": agones.dev/agones/test/e2e/extensions.TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown(0xc0002f6d00)
generic-1.24: Step #1 - "e2e-feature-gates": 	/go/src/agones.dev/agones/test/e2e/extensions/high_availability_test.go:56 +0x490
generic-1.24: Step #1 - "e2e-feature-gates": testing.tRunner(0xc0002f6d00, 0x258d3f0)
generic-1.24: Step #1 - "e2e-feature-gates": 	/usr/local/go/src/testing/testing.go:1446 +0x217
generic-1.24: Step #1 - "e2e-feature-gates": created by testing.(*T).Run
generic-1.24: Step #1 - "e2e-feature-gates": 	/usr/local/go/src/testing/testing.go:1493 +0x75e
generic-1.24: Step #1 - "e2e-feature-gates": FAIL test/e2e/extensions.TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown (0.15s)
generic-1.24: Step #1 - "e2e-feature-gates": FAIL test/e2e/extensions
```

**Comment by markmandel** [comment]
Spent an hour and 30 minutes waiting to be leader.

`e2e-wait-to-become-leader- | 01:30:00-- | --`

@zmerlynn seeing this a bit these days? Just too many builds?

**Comment by markmandel** [comment]
Got this flake again:

```
gke-autopilot-1.25: Step #1 - "e2e-feature-gates": time="2023-02-23 20:09:58.528" level=info msg="Removing one of the Extensions Pods" test=TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown
gke-autopilot-1.25: Step #1 - "e2e-feature-gates": time="2023-02-23 20:09:58.605" level=info msg="Creating default game server" test=TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown
gke-autopilot-1.25: Step #1 - "e2e-feature-gates":     high_availability_test.go:55: 
gke-autopilot-1.25: Step #1 - "e2e-feature-gates":         	Error Trace:	/go/src/agones.dev/agones/test/e2e/extensions/high_availability_test.go:55
gke-autopilot-1.25: Step #1 - "e2e-feature-gates":         	Error:      	Received unexpected error:
gke-autopilot-1.25: Step #1 - "e2e-feature-gates":         	            	creating {game-server [{udp-port Dynamic <nil> 7654 0 UDP}] {false 0 0 0}  { 0 0} {{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []} {[] [] [{game-server us-docker.pkg.dev/agones-images/examples/simple-game-server:0.14 [] []  [] [] [] {map[cpu:{{30 -3} {<nil>} 30m DecimalSI} memory:{{33554432 0} {<nil>}  BinarySI}] map[cpu:{{30 -3} {<nil>} 30m DecimalSI} memory:{{33554432 0} {<nil>}  BinarySI}]} [] [] nil nil nil nil   IfNotPresent nil false false false}] []  <nil> <nil>  map[]   <nil>  false false false <nil> nil []   nil  [] []  <nil> nil [] <nil> <nil> <nil> map[] [] <nil> nil}} <nil> map[] map[] <nil>} GameServer instances failed (): Internal error occurred: failed calling webhook "mutations.agones.dev": failed to call webhook: Post "https://agones-controller-service.agones-system.svc:443/mutate?timeout=10s": dial tcp 10.12.1.118:8081: connect: connection refused
gke-autopilot-1.25: Step #1 - "e2e-feature-gates":         	Test:       	TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown
gke-autopilot-1.25: Step #1 - "e2e-feature-gates":         	Messages:   	Could not get a GameServer ready
gke-autopilot-1.25: Step #1 - "e2e-feature-gates": --- FAIL: TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown (0.18s)
gke-autopilot-1.25: Step #1 - "e2e-feature-gates": panic: runtime error: invalid memory address or nil pointer dereference [recovered]
gke-autopilot-1.25: Step #1 - "e2e-feature-gates": 	panic: runtime error: invalid memory address or nil pointer dereference
gke-autopilot-1.25: Step #1 - "e2e-feature-gates": [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x20fa2b0]
gke-autopilot-1.25: Step #1 - "e2e-feature-gates": 
```

**Pull request merged**